lib/src/core/models/functions/functions.dart

```diff
@@ -13,83 +13,190 @@ class OpenAIFunctionModel {
 
   /// The parameters the functions accepts, described as a
   /// [JSON Schema](https://json-schema.org/understanding-json-schema) object.
-  final OpenAIFunctionParameters? parameters;
+  final Map<String, dynamic> parametersSchema;
 
   const OpenAIFunctionModel({
     required this.name,
+    required this.parametersSchema,
     this.description,
-    this.parameters,
   });
 
+  factory OpenAIFunctionModel.withParameters({
+    required String name,
+    String? description,
+    required Iterable<OpenAIFunctionProperty> parameters,
+  }) {
+    return OpenAIFunctionModel(
+      name: name,
+      description: description,
+      parametersSchema: OpenAIFunctionProperty.object(
+        name: '',
```

Comment:
Awesome, but this name: "" is weird to see it like this, can we do something about it